src/form-builder/types.ts

```diff
@@ -1,6 +1,6 @@
 import { FieldHookConfig } from 'formik';
 
-export type FieldProps<V> = FieldHookConfig<V> & {
+export type FieldProps<V> = Omit<FieldHookConfig<V>, 'required'> & {
```

Comment:
Turns out intersecting types (`&`) uses the least common denominator between joined types (I believe) on a property. So `{ required?: boolean } & { required?: boolean | string }` --> `{ required: boolean | undefined }`.

To get around this, I'm omitting the original `required` property and adding it back in on the intersecting type.

Reply:
Maybe it's because I'm a typescript noob, but this is a bit confusing to me. Can you add a comment?